pkg/kyma/cmd/test/cmd.go

```diff
@@ -0,0 +1,155 @@
+	"github.com/kyma-incubator/kymactl/internal/step"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/helm/pkg/helm"
+	"k8s.io/helm/pkg/helm/environment"
+	helm_release "k8s.io/helm/pkg/proto/hapi/release"
+)
+
+type Options struct {
+	step.Factory
+	skip []string
+	verbose bool
+}
+
+var testNamespaces = []string{
+	"kyma-system",
+	"istio-system",
+	"knative-serving",
+	"kyma-integration",
+}
+
+var testReleases = map[string]bool{
+	"core": true,
+	"monitoring": false,
+	"logging": false,
+	"istio": true,
+	"knative": true,
+	"application-connector": true,
+}
+
+//NewCmd creates a new install command
+func NewCmd() *cobra.Command {
+	opts := &Options{}
+	cmd := &cobra.Command{
+		Use:   "test",
+		Short: "test kyma installation",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringArrayVar(&opts.skip, "skip", []string{}, "Skip tests for these releases")
+	cmd.Flags().BoolVar(&opts.NonInteractive, "non-interactive", false, "Do not use spinners")
+	cmd.Flags().BoolVarP(&opts.verbose, "verbose", "v", false, "Print additional output")
+	return cmd
+}
+
+func (opts *Options) Run() error {
+	helmConfig := &environment.EnvSettings{TillerConnectionTimeout: 300}
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", clientcmd.RecommendedHomeFile)
+	if err != nil {
+		return err
+	}
+
+	helmClient, err := kyma_helm.New(helmConfig, kubeConfig)
+	if err != nil {
+		return err
+	}
+
+	for _, ns := range testNamespaces {
+		err := opts.cleanHelmTestPods(ns)
+		if err != nil {
+			return err
+		}
+	}
+
+	for release := range testReleases {
+		s := opts.NewStep(
+			fmt.Sprintf("Testing release %s", release),
+		)
+		s.Start()
+
+		if opts.skipRelease(helmClient, release) {
+			s.Successf("Skipping release %s", release)
+			continue
+		}
+
+		msg, success, err := opts.testRelease(helmClient, s, release)
+		s.Stop(success)
+		if msg != "" && (!success || opts.verbose) {
+			fmt.Println("--- Log ---")
+			fmt.Print(msg)
+			fmt.Println("---")
+		}
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (opts *Options) skipRelease(client helm.Interface, release string) bool {
+	for _, skipped := range opts.skip {
+		if skipped == release {
+			return true
+		}
+	}
+
+	if !testReleases[release] {
+		_, err := client.ReleaseStatus(release)
+		if err != nil {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (opts *Options) cleanHelmTestPods(namespace string) error {
+	s := opts.NewStep(
+		fmt.Sprintf("Cleaning up helm test pods in namespace %s", namespace),
+	)
+	s.Start()
+
+	_, err := internal.RunKubectlCmd([]string{"delete", "pod", "-n", namespace, "-l", "helm-chart-test=true"})
+	if err != nil {
+		s.Failure()
+		return err
+	}
+
+	s.Success()
+	return nil
+}
+
+func (opts *Options) testRelease(client helm.Interface, s step.Step, release string) (string, bool, error) {
+	c, errc := client.RunReleaseTest(release, helm.ReleaseTestTimeout(600))
```

Comment:
After I executed all tests (with failing knative release) I tried to re-run the test command. It passed all stages again but was not executing any test anymore. Please re-check

Reply:
This is probably because test pods were not removed before tests. I need to investigate further.

Reply:
I wasn't able to reproduce after I made changes with reading components from installation CR. Could you recheck on new version?

Reply:
fine for me now.